spyder/plugins/ipythonconsole/widgets/main_widget.py

```diff
@@ -2301,16 +2305,27 @@ def run_script(self, filename, wdir, args, debug, post_mortem,
                     line += ", current_namespace=True"
                 line += ")"
 
-                if debug:
+                if method == "debugfile":
                     # To keep focus in editor after running debugfile
                     client.shellwidget._pdb_focus_to_editor = focus_to_editor
-            else:  # External, non spyder-kernels, use %run
+            elif method in ["runfile", "debugfile"]:
+                # External, non spyder-kernels, use %run
                 line = "%run "
-                if debug:
+                if method == "debugfile":
                     line += "-d "
                 line += "\"%s\"" % str(filename)
                 if args:
                     line += " %s" % norm(args)
+            else:
+                QMessageBox.warning(
+                    self,
+                    _('Warning'),
+                    _("The client is not a spyder-kernel "
+                      "to run <b>{}</b>.<br><br>Please Use a spyder-kernel."
+                      ).format(method),
+                    QMessageBox.Ok
+                )
+                return
```

Comment:
I don't understand why you added this. Precisely the previous `elif` takes care of non-Spyder kernels, as the comment below it clearly says.

Reply:
The reason is that I plan to add a profile_file function (and maybe others) so if it is a spyder-kernels, no problem, if runfile, we can run on ipython, if debug file, we can run with -d. For profile file, nothing works

Reply:
```suggestion
            else:
                client.shellwidget.append_html_message(
                    _("The console is not running a Spyder-kernel, so it "
                      "can't execute <b>{}</b>.<br><br>"
                      "Please use a Spyder-kernel for this.").format(method),
                    msg_type='error',
                    before_prompt=True
                )
                return
```

Ok, thanks for the clarification @impact27. Then I suggest using the API for messages in `shellwidget` for this.